mbed-events/EventQueue.h

```diff
@@ -0,0 +1,764 @@
+            f(a0, a1, a2, a3, b0, b1, b2, b3);
+        }
+    };
+
+    template <typename F, typename A0, typename A1, typename A2, typename A3, typename A4, typename B0, typename B1, typename B2, typename B3>
+    struct context54 {
+        F f; A0 a0; A1 a1; A2 a2; A3 a3; A4 a4;
+        context54(F f, A0 a0, A1 a1, A2 a2, A3 a3, A4 a4)
+            : f(f), a0(a0), a1(a1), a2(a2), a3(a3), a4(a4) {}
+
+        void operator()(B0 b0, B1 b1, B2 b2, B3 b3) {
+            f(a0, a1, a2, a3, a4, b0, b1, b2, b3);
+        }
+    };
+
+    template <typename F, typename B0, typename B1, typename B2, typename B3, typename B4>
+    struct context05 {
+        F f;
+        context05(F f)
+            : f(f) {}
+
+        void operator()(B0 b0, B1 b1, B2 b2, B3 b3, B4 b4) {
+            f(b0, b1, b2, b3, b4);
+        }
+    };
+
+    template <typename F, typename A0, typename B0, typename B1, typename B2, typename B3, typename B4>
+    struct context15 {
+        F f; A0 a0;
+        context15(F f, A0 a0)
+            : f(f), a0(a0) {}
+
+        void operator()(B0 b0, B1 b1, B2 b2, B3 b3, B4 b4) {
+            f(a0, b0, b1, b2, b3, b4);
+        }
+    };
+
+    template <typename F, typename A0, typename A1, typename B0, typename B1, typename B2, typename B3, typename B4>
+    struct context25 {
+        F f; A0 a0; A1 a1;
+        context25(F f, A0 a0, A1 a1)
+            : f(f), a0(a0), a1(a1) {}
+
+        void operator()(B0 b0, B1 b1, B2 b2, B3 b3, B4 b4) {
+            f(a0, a1, b0, b1, b2, b3, b4);
+        }
+    };
+
+    template <typename F, typename A0, typename A1, typename A2, typename B0, typename B1, typename B2, typename B3, typename B4>
+    struct context35 {
+        F f; A0 a0; A1 a1; A2 a2;
+        context35(F f, A0 a0, A1 a1, A2 a2)
+            : f(f), a0(a0), a1(a1), a2(a2) {}
+
+        void operator()(B0 b0, B1 b1, B2 b2, B3 b3, B4 b4) {
+            f(a0, a1, a2, b0, b1, b2, b3, b4);
+        }
+    };
+
+    template <typename F, typename A0, typename A1, typename A2, typename A3, typename B0, typename B1, typename B2, typename B3, typename B4>
+    struct context45 {
+        F f; A0 a0; A1 a1; A2 a2; A3 a3;
+        context45(F f, A0 a0, A1 a1, A2 a2, A3 a3)
+            : f(f), a0(a0), a1(a1), a2(a2), a3(a3) {}
+
+        void operator()(B0 b0, B1 b1, B2 b2, B3 b3, B4 b4) {
+            f(a0, a1, a2, a3, b0, b1, b2, b3, b4);
+        }
+    };
+
+    template <typename F, typename A0, typename A1, typename A2, typename A3, typename A4, typename B0, typename B1, typename B2, typename B3, typename B4>
+    struct context55 {
+        F f; A0 a0; A1 a1; A2 a2; A3 a3; A4 a4;
+        context55(F f, A0 a0, A1 a1, A2 a2, A3 a3, A4 a4)
+            : f(f), a0(a0), a1(a1), a2(a2), a3(a3), a4(a4) {}
+
+        void operator()(B0 b0, B1 b1, B2 b2, B3 b3, B4 b4) {
+            f(a0, a1, a2, a3, a4, b0, b1, b2, b3, b4);
+        }
+    };
+};
+
+}
+
+
+// Include event class here to workaround cyclic dependencies
```

Comment:
There are a few other changes requested in this PR that are related to `mbed-events`, so it makes sense to also address them upstream. After we agree that all changes to `mbed-events` are in, I'll update this PR.